tests/marine/test_marine.py

```diff
@@ -2,85 +2,217 @@
-                              extracted_fields_from_udp_packet):
-    expected = dict(zip(extracted_fields_from_udp_packet, map(str, [mac_1, mac_2, ip_1, ip_2, port_3, port_4])))
-    passed, output = marine_instance.filter_and_parse(udp_packet, 'ip', 'udp', extracted_fields_from_udp_packet)
+from pytest_lazyfixture import lazy_fixture
+from parametrization import Parametrization
+from typing import List, Union
+from marine import Marine
+
+
+# TODO: Add a test for FTP.
+
+
+@Parametrization.parameters(
+    "packet",
+    "fields_expected_fixture_name",
+    "extracted_fields_from_packet",
+    "bpf_filter",
+    "display_filter",
+)
+@Parametrization.case(
+    name="ARP packet",
+    packet=lazy_fixture("arp_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "broadcast_mac",
+        "mac_1",
+        "ip_1",
+        "broadcast_mac",
+        "ip_2",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_arp_packet"),
+    bpf_filter="arp",
+    display_filter="arp",
+)
+@Parametrization.case(
+    name="ICMP packet",
+    packet=lazy_fixture("icmp_packet"),
+    fields_expected_fixture_name=["mac_1", "mac_2", "ip_1", "ip_2", "icmp_echo_type"],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_icmp_packet"),
+    bpf_filter="ip",
+    display_filter="icmp",
+)
+@Parametrization.case(
+    name="TCP packet",
+    packet=lazy_fixture("tcp_packet"),
+    fields_expected_fixture_name=["mac_1", "mac_2", "ip_1", "ip_2", "port_1", "port_2"],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_tcp_packet"),
+    bpf_filter="ip",
+    display_filter="tcp",
+)
+@Parametrization.case(
+    name="DNS packet",
+    packet=lazy_fixture("dns_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "ip_2",
+        "port_3",
+        "dns_port",
+        "url_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_dns_packet"),
+    bpf_filter="ip",
+    display_filter="dns",
+)
+@Parametrization.case(
+    name="DHCP packet",
+    packet=lazy_fixture("dhcp_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "broadcast_ip",
+        "port_3",
+        "dhcp_port",
+        "ip_2",
+        "ip_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_dhcp_packet"),
+    bpf_filter="ip",
+    display_filter="dhcp",
+)
+@Parametrization.case(
+    name="HTTP packet",
+    packet=lazy_fixture("http_packet"),
+    fields_expected_fixture_name=[
+        "mac_1",
+        "mac_2",
+        "ip_1",
+        "ip_2",
+        "port_1",
+        "http_port",
+        "http_type",
+        "http_uri",
+        "http_version",
+        "url_1",
+    ],
+    extracted_fields_from_packet=lazy_fixture("extracted_fields_from_http_packet"),
+    bpf_filter="ip",
+    display_filter="http",
+)
+def test_packet_filter_and_parse(
```

Comment:
The parametrization for this test is way too complicated for what it tests.
I'd really rather you avoided fixtures in this test.
I want to see every tested packet next to every field and its value (with only relevant fields).
Something like:
```python
@Parametrization.case(
    name="HTTP",
    packet=Ethernet(...) + IP(...) + TCP(...) + HTTP(...),
    bpf="ip",
    display_filter="http",
    fields=["http.type", "http.uri", "http.version"],
    expected=["GET", "/pasten", "1.1"],
)
```
Perhaps with some predefined consts for reptetive and *not interesting* layers (not interesting = not tested).

Reply:
I think creating the packet for each case in the file will create a bit of a mess [for example, look how many lines are needed to create a DHCP packet](https://github.com/tomlegkov/marine-python/pull/8/files#diff-53b15373e3037da4b4e57bb3b36811dcR247).
Is it really interesting to see the values being tested when you read the case, especially when there's only one packet per case? Hiding a lot of the data behind fixtures makes the case a lot more readable.

Reply:
I'm not sure about it. My instinct is to make tests as explicit as possible.
Packets with extremly long tested layer (such as DHCP) might indeed justify a fixture (or const), but a standalone one, with no long dependency graph like now.

Reply:
I understand your instinct and I agree with it, however it means some cases will contain fixtures and some won't. I think it will be inconvenient to follow.

Maybe the solution is to go back to not using parameterize? Having a separate test for every protocol will allow more flexibility, but will contain code repetition.